python/delta/tests/test_deltatable.py

```diff
@@ -698,6 +698,20 @@ def test_create_table_with_existing_schema(self) -> None:
                                        nullables={"key", "value", "value2"},
                                        partitioningColumns=["value", "value2"])
 
+        with self.table("test3"):
```

Comment:
Let's use a separate function, maybe `test_create_table_with_cluster_by`, and let's also test replace table

Reply:
Made `test_create_replace_table_with_cluster_by`